react/features/whiteboard/components/web/WhiteboardButton.tsx

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable require-jsdoc */
+/* eslint-disable lines-around-comment */
+import type { Dispatch } from 'redux';
+
+import { IStateful } from '../../../base/app/types';
+// @ts-ignore
+import { translate } from '../../../base/i18n';
+// @ts-ignore
+import { IconShareAudio, IconStopAudioShare } from '../../../base/icons';
+// @ts-ignore
+import { connect } from '../../../base/redux';
+// @ts-ignore
+import { AbstractButton, type AbstractButtonProps } from '../../../base/toolbox/components';
+// @ts-ignore
+import { setOverflowMenuVisible } from '../../../toolbox/actions.web';
+import { getWhiteboardButtonProps } from '../../functions';
+
+
+type Props = AbstractButtonProps & {
+
+    /**
+     * The callback to execute for click events.
+     */
+    _action: Function;
+
+    /**
+     * Whether or not the button is toggled.
+     */
+    _toggled: boolean;
+
+    /**
+     * Whether or not the whiteboard is visible.
+     */
+    _visible: boolean;
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Dispatch<any>;
+};
+
+/**
+ * Component that renders a toolbar button for the whiteboard.
+ */
+class WhiteboardButton extends AbstractButton<Props, any, any> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.whiteboard';
+    icon = IconShareAudio; // TODO: change icons
+    label = 'toolbar.showWhiteboard';
+    toggledIcon = IconStopAudioShare;
+    toggledLabel = 'toolbar.hideWhiteboard';
+    toggledTooltip = 'toolbar.hideWhiteboard';
+    tooltip = 'toolbar.showWhiteboard';
+
+    /**
+     * Handles clicking / pressing the button, and opens / closes the whiteboard view.
+     *
+     * @private
+     * @returns {void}
+     */
+    _handleClick() {
+
+        // @ts-ignore
+        const { dispatch, _action } = this.props;
+
+        dispatch(_action());
```

Comment:
WAT

Reply:
show/hide from the props, depending on the context, what?